app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -8,132 +8,224 @@ import TSnePlotViewRoute from './TSnePlotViewRoute'
 import ExperimentDesignRoute from './ExperimentDesignRoute'
 import SupplementaryInformationRoute from './SupplementaryInformationRoute'
 import DownloadsRoute from './DownloadsRoute'
+import {isEmptyArray, tabCommonValidations, tabValidations} from "./TabConfig";
 
 const RoutePropTypes = {
-  match: PropTypes.object.isRequired,
-  location: PropTypes.object.isRequired,
-  history: PropTypes.object.isRequired
+    match: PropTypes.object.isRequired,
+    location: PropTypes.object.isRequired,
+    history: PropTypes.object.isRequired
 }
 
 const TabCommonPropTypes = {
-  atlasUrl: PropTypes.string.isRequired,
-  experimentAccession: PropTypes.string.isRequired,
-  species: PropTypes.string.isRequired,
-  accessKey: PropTypes.string,
-  resourcesUrl: PropTypes.string
+    atlasUrl: PropTypes.string.isRequired,
+    experimentAccession: PropTypes.string.isRequired,
+    species: PropTypes.string.isRequired,
+    accessKey: PropTypes.string,
+    resourcesUrl: PropTypes.string
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
 const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
+    'results' : TSnePlotViewRoute,
+    'experiment-design' : ExperimentDesignRoute,
+    'supplementary-information' : SupplementaryInformationRoute,
+    'downloads' : DownloadsRoute
 }
 
-const TopRibbon = ({tabs, routeProps}) =>
-  <ul className={`tabs`}>
-    {
-      tabs.map((tab) =>
-        <li title={tab.name} key={tab.type} className={`tabs-title`}>
-          <NavLink to={{pathname:`/${tab.type}`, search: routeProps.location.search, hash: routeProps.location.hash}}
-                   activeClassName={`active`}>
-            {tab.name}
-          </NavLink>
-        </li>
-      )}
-  </ul>
+function shouldRenderTab(tab, commonProps) {
+    const commonRequiredProps = tabCommonValidations.get(tab.type);
```

Comment:
Could you tell me please why we need this validation?
You have already defined the required properties in the `PropTypes`.
Could you please rework or even remove this whole `shouldRender` method, if you agree with me?
Thanks

Reply:
Ok, we discussed this why you wrote it. I am going to resolve this issue and review this code again.